internal/provider/resource_consul_cluster.go

```diff
@@ -767,3 +832,37 @@ func resourceConsulClusterImport(ctx context.Context, d *schema.ResourceData, me
 
 	return []*schema.ResourceData{d}, nil
 }
+
+// buildIPAllowlist validates there are no duplicate CIDRs and returns a consul model.
+func buildIPAllowlist(cidrs []interface{}) ([]*consulmodels.HashicorpCloudConsul20210204CidrRange, error) {
+	var IPAllowList []*consulmodels.HashicorpCloudConsul20210204CidrRange
+	if len(cidrs) == 0 {
+		return IPAllowList, nil
+	}
+
+	// Seen holds validated CIDRs.
+	seen := make(map[string]struct{}, len(cidrs))
+	for _, cidr := range cidrs {
+		cidrMap := cidr.(map[string]interface{})
+		address := cidrMap["address"].(string)
+		description := cidrMap["description"].(string)
+
+		ip, _, _ := net.ParseCIDR(address)
```

Comment:
Validation (validators.go) occurs first , so this should be valid (no error). I omitted handling it, but happy to do it if this feels like bad practice.